source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRChoiceSetViewDataSource.mm

```diff
@@ -24,17 +21,6 @@ - (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(nulla
 {
     self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
     if(self) {
-        UIImage *iconImage = nil;
```

Comment:
this initWithStyle method gets called when dequeueReusableCellWithIdentifier is called and it can't get reusable cell.  This ensures that the cell with the correct image will get reused, and the code changes only the label. With your change, we are loading and setting the image each time cell is asked.

Since initWithStyle is get called by dequeReusableCellWithIdentifier, and thus we can't pass our resource resolver, you will have to add a method to ACRChoiceSetCell that takes resource resolver as argument. and you call this method once the deque call return on the returned cell. 
inside the method, you can simply check if icon image's.imge is set, if not, we load it there.

Reply:
What about storing the images as properties and use them ?